File: Script4Test/swift/Script4Test.swift
// Только тело функции становится блоками; шапка переносится как есть.
// Внутри — письменное подмножество NSG.
func crunch(_ n: Int) -> Int {
    var primes: Int = 0
    var i: Int = 2
    while i <= n {
        var isPrime: Int = 1
        var j: Int = 2
        while j * j <= i {
            if i % j == 0 {
                isPrime = 0
            }
            j += 1
        }
        primes += isPrime
        i += 1
    }
    let mask: Int = (1 << 8) - 1
    let folded: Int = (primes & mask) ^ (primes >> 3)
    let bonus: Int = folded > 0 ? folded : -1
    return primes + bonus
}

File: Script4Test/python/script4test.py
# Только тело функции становится блоками; шапка переносится как есть.
# Внутри — письменное подмножество NSG.
def crunch(n):
    primes = 0
    i = 2
    while i <= n:
        is_prime = 1
        j = 2
        while j * j <= i:
            if i % j == 0:
                is_prime = 0
            j += 1
        primes += is_prime
        i += 1
    mask = (1 << 8) - 1
    folded = (primes & mask) ^ (primes >> 3)
    bonus = -1
    if folded > 0:
        bonus = folded
    return primes + bonus
